src/main.zig

```diff
@@ -2207,6 +2208,12 @@ pub fn cmdBuild(gpa: *Allocator, arena: *Allocator, args: []const []const u8) !v
                             });
                             fatal("No 'build.zig' file found, in the current directory or any parent directories.", .{});
                         };
+
+                        //Avoid getting stuck in loop when at root.
+                        if (std.mem.eql(u8, dirname, prevDirname))
```

Comment:
`std.fs.path.dirname` does not return a reallocated slice, so a simple `dirname.len != prevDirname.len` should be sufficient here and a lot faster. See the implementation for dirname:
https://github.com/ziglang/zig/blob/master/lib/std/fs/path.zig#L752-L820

Reply:
I'll try that out, will test some edge cases like parent child dirs being same length

Reply:
To my understanding, it should never happen, as the returned string is only a part of the string passed in. It can never get any longer and if it has the same length, it *must* be identical

Reply:
Updated it to use len of slice as suggested by @MasterQ32, works for edge cases